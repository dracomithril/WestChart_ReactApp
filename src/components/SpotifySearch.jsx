/**
 * Created by Gryzli on 09.04.2017.
 */
import React from "react";
import PropTypes from "prop-types";
import PlaylistForm from "./PlaylistForm";
import RowSpotifySearch from "./RowSpotifySearch"


export default class SpotifySearch extends React.Component {
    /*istanbul ignore next*/
    componentWillUnmount() {
        console.log('component SpotifySearch unmounted');
    }

    /*istanbul ignore next*/
    componentDidMount() {
        console.log('component SpotifySearch did mount');
    }

    render() {
        const {store} = this.context;
        const {search_list, sp_playlist_info} = store.getState();
        let search_list_view = search_list.map((search_elem)=> {
            return <RowSpotifySearch search_elem={search_elem} key={search_elem.id}/>
        });
        return (<div>
            <h3 id="list">{'Search music by: '}
                {sp_playlist_info.url !== null && <div className="spotify_sumary">
                    <span>{'Created playlist! name: ' + sp_playlist_info.pl_name}</span><br/>
                    <a href={sp_playlist_info.url} target="_newtab">{sp_playlist_info.pl_name}</a>
                </div>}
                <PlaylistForm {...this.props}/>
            </h3>
            {search_list_view.length > 0 && <table style={{marginLeft:10}}>
                <thead>
                <tr>
                    <th>Artist & Title</th>
                    <th>Options:</th>
                    <th>Selected:</th>
                </tr>
                </thead>
                <tbody>
                {search_list_view}
                </tbody>
            </table>}
        </div>);
    }
}
SpotifySearch.contextTypes = {
    store: PropTypes.object
};
SpotifySearch.propTypes = {
    selected: PropTypes.array
};