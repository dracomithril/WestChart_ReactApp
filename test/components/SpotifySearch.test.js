/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import SpotifySearch from '../../src/front/components/SpotifySearch';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
const initial_state =require('../data/initial_state.json');

const mockStore = configureMockStore([]);
describe('<SpotifySearch/>', () => {
    it('renders without crashing', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <SpotifySearch selected={[]}/>, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();

    });
});
