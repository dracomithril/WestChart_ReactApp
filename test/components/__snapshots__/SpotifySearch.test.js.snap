// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpotifySearch/> renders without crashing 1`] = `
<div>
  <h3
    id="list"
  >
    Create 
    <i
      aria-hidden="true"
      className="fa fa-spotify"
    >
      Spotify playlist:
    </i>
    <PlaylistForm
      selected={Array []}
    />
  </h3>
</div>
`;
